### PR TITLE
Implementing vale 

### PR DESCRIPTION
**Pull request by RobPasMue**
As title says and coming from request on #249 by @PipKat 

I'll leave you the error resolutions! =)

### CODE REVIEW

**Comment by PipKat** [comment]
@RobPasMue All remaining Value issues can be resolved by excluding this folder from the Granta MI BomAnalaytics Vale check: C:\AnsysDev\GitRepos\PyAnsys\grantami-bomanalytics\doc\source\_templates. I tried adding "-templates" to exclude_patterns in the config.py file but that didn't work. Also @Andy-Grigg, primariliy so that you know that Vale is implemented and only removing the _templates folder from the check is left to do.

**Comment by Andy-Grigg** [comment]
Thanks for implementing this! I'm (mostly) off today and Monday, but I'll get this finished off next week.

**Comment by RobPasMue** [comment]
Hey there! @Andy-Grigg @PipKat - I briefly discussed with Kathy, the way Vale works is basically "the other way around". You tell the tool which files to consider. And the way we have it configured in PyAnsys actions is to consider the entire "doc" folder. If you want to try and work around it, I suggest we implement the action "on our own" https://github.com/pyansys/actions/blob/main/doc-style/action.yml

This might be a bit tricky though... wouldn't it be easier to just solve the issues in the templates or ignore Vale in the files with the use of the pragmas ``.. vale off / .. vale on```? Depending on:
* If it is an HTML file or Markdown: https://vale.sh/docs/topics/config/#markdown-amp-html
* If it is an rST file: https://vale.sh/docs/topics/config/#restructuredtext

**Comment by PipKat** [comment]
@Andy-Grigg I implemented .. vale off/.. vale in the RST files in the _templates folder. There are no longer Vale errors. I've approved but will let you decide if you want to merge.

**Review by PipKat — APPROVED** [review]
All Vale errors now resolved. LGTM.

File: src/ansys/grantami/bomanalytics/queries.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def __init__(self) -> None:
     @property
     def populated_inputs(self) -> bool:
         """Whether the argument manager is populated. For example, this property
-        determinse whether a query be performed on the items in the object.
+        determines whether a query be performed on the items in the object.
```
Comment (PipKat) [inline src/ansys/grantami/bomanalytics/queries.py:84]:
```suggestion
        determines whether to perform a query on the items in the object.
```

**Review by Andy-Grigg — CHANGES_REQUESTED** [review]

File: examples/0_Getting_started.py [hunk 2]
```diff
@@ -93,7 +93,11 @@
 # consolidate the cells above into a single step:
 
 # + tags=[]
-query = queries.MaterialImpactedSubstancesQuery().with_material_ids(["plastic-abs-high-impact"]).with_legislations(["REACH - The Candidate List"])  # noqa: E501
+query = (
```
[the remark this answers is not in the document]
Comment (PipKat) [reply examples/0_Getting_started.py]: @Andy-Grigg I THINK I restored it correctly. Please resolve the conversation if so.

File: examples/0_Getting_started.py [hunk 3]
```diff
@@ -93,7 +93,7 @@
 # consolidate the cells above into a single step:
 
 # + tags=[]
-query = queries.MaterialImpactedSubstancesQuery().with_material_ids(["plastic-abs-high-impact"]).with_legislations(["REACH - The Candidate List"])  # noqa: E501
+query = queries.MaterialImpactedSubstancesQuery().with_material_ids(["plastic-abs-high-impact"]).with_legislations(["REACH - The Candidate List"]))  # noqa: E501
```
Comment (PipKat) [inline examples/0_Getting_started.py:96]:
```suggestion
query = queries.MaterialImpactedSubstancesQuery().with_material_ids(["plastic-abs-high-impact"]).with_legislations(["REACH - The Candidate List"])  # noqa: E501
```

**Review by Andy-Grigg — APPROVED** [review]

**Pull request merged**